app/celery/tasks.py

```diff
@@ -635,49 +632,57 @@ def update_letter_notifications_to_error(self, notification_references):
+    # SQLAlchemy is throwing a FlushError. So we check if the notification id already exists then do not
+    # send to the retry queue.
+    found = get_notification_by_id(notification_id)
+    if not found:
+        retry_msg = "{task} notification for job {job} row number {row} and notification id {notif}".format(
             task=task.__name__,
             job=notification.get("job", None),
             row=notification.get("row_number", None),
-            noti=notification_id,
+            notif=notification_id,
         )
-        # Sometimes, SQS plays the same message twice. We should be able to catch an IntegrityError, but it seems
-        # SQLAlchemy is throwing a FlushError. So we check if the notification id already exists then do not
-        # send to the retry queue.
         current_app.logger.exception("Retry" + retry_msg)
         try:
-            task.retry(queue=QueueNames.RETRY, exc=exc)
+            task.retry(queue=QueueNames.RETRY, exc=exception)
         except task.MaxRetriesExceededError:
             current_app.logger.error("Max retry failed" + retry_msg)
 
 
-def handle_list_of_exception(task, list_notification, exc, receipt: Optional[UUID]):
+def handle_batch_error_and_forward(
+    task, signed_and_verified: list[tuple[Any, Any]], notification_type: str, exception, receipt: UUID = None
+):
     if receipt:
-        current_app.logger.info(f"Batch saving: could not persist notifications with receipt {receipt}")
+        current_app.logger.exception(f"Batch saving: could not persist notifications with receipt {receipt}", exception)
     else:
-        current_app.logger.info("Batch saving: could not persist notifications.")
-    for notification in list_notification:
+        current_app.logger.exception("Batch saving: could not persist notifications.", exception)
+
+    for (signed, notification) in signed_and_verified:
         notification_id = notification["notification_id"]
-        if not get_notification_by_id(notification_id):
-            retry_msg = (
-                "{task} notification for job {job} row number {row} and notification id {notif} and receipt {receipt}".format(
-                    task=task.__name__,
-                    job=notification.get("job", None),
-                    row=notification.get("row_number", None),
-                    notif=notification_id,
-                    receipt=receipt,
-                )
+        service = notification["service"]
+        # Sometimes, SQS plays the same message twice. We should be able to catch an IntegrityError, but it seems
+        # SQLAlchemy is throwing a FlushError. So we check if the notification id already exists then do not
+        # send to the retry queue.
+        found = get_notification_by_id(notification_id)
+        if not found and service:
+            forward_msg = "Batch saving: forwarding notification {notif} to individual save from receipt {receipt}.".format(
```

Comment:
At this point, do you want to `break` from the loop so the `else` statement below doesn't execute?

Reply:
Yep!